cmd/kops/create_secret_cacert.go

```diff
@@ -0,0 +1,144 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+
+	"github.com/golang/glog"
+	"github.com/spf13/cobra"
+
+	"k8s.io/kops/cmd/kops/util"
+	"k8s.io/kops/pkg/pki"
+	"k8s.io/kops/upup/pkg/fi"
+	"k8s.io/kops/upup/pkg/fi/utils"
+	"k8s.io/kubernetes/pkg/kubectl/cmd/templates"
+	"k8s.io/kubernetes/pkg/kubectl/util/i18n"
+)
+
+var (
+	createSecretCacertLong = templates.LongDesc(i18n.T(`
+	Add a ca certificate and private key.
+    `))
+
+	createSecretCacertExample = templates.Examples(i18n.T(`
+	Add a ca certificate and private key.
+	kops create secret ca-cert \
+		--cert ~/ca.pem \
+		--key ~/ca-key.pem \
+		--name k8s-cluster.example.com \
+		--state s3://example.com
+	`))
+
+	createSecretCacertShort = i18n.T(`Add a ca cert and key`)
+)
+
+type CreateSecretCaCertOptions struct {
+	ClusterName    string
+	PrivateKeyPath string
+	CertPath       string
+}
+
+// NewCmdCreateSecretCaCert returns create ca certificate command
+func NewCmdCreateSecretCaCert(f *util.Factory, out io.Writer) *cobra.Command {
+	options := &CreateSecretCaCertOptions{}
+
+	cmd := &cobra.Command{
+		Use:     "ca-cert",
+		Short:   createSecretCacertShort,
+		Long:    createSecretCacertLong,
+		Example: createSecretCacertExample,
+		Run: func(cmd *cobra.Command, args []string) {
+			err := rootCommand.ProcessArgs(args)
+			if err != nil {
+				exitWithError(err)
+			}
+
+			options.ClusterName = rootCommand.ClusterName()
+
+			err = RunCreateSecretCaCert(f, os.Stdout, options)
+			if err != nil {
+				exitWithError(err)
+			}
+		},
+	}
+
+	cmd.Flags().StringVar(&options.CertPath, "cert", options.CertPath, "Path to ca cert")
+	cmd.Flags().StringVar(&options.PrivateKeyPath, "key", options.PrivateKeyPath, "Path to ca cert private key")
```

Comment:
Technically this should probably be "Path to CA private key" (and "Path to CA certificate") to avoid the awkwardness of CA cert private key :-)

Reply:
makes since